woocommerce/wc-admin
# Add header cards for all tasks in Tasklist UI experiment

Fixes https://github.com/woocommerce/woocommerce-admin/issues/7636

This PR:

1. Adds header cards for all the tasks in the Tasklist UI experiment.
2. Changes the logic for `woocommerce-payments` task to revert back to `payments` task after it has been completed. (Based on this [comment](https://github.com/woocommerce/woocommerce-admin/issues/7636#issuecomment-948515677)).
3. Updates tracking to behave similarly to the existing task list.
3. Adds react-powered SVG components to enable CSS style customization.

### Screenshots

![image](https://user-images.githubusercontent.com/3747241/138540779-71089834-f22b-431a-aba4-0f0161ffabd9.png)


### Detailed test instructions:

**Pre-requisites**
1. In a fresh site, go through the setup wizard, make sure **NOT** to install WCPay in Business Features step.
2. Make sure tracking is enabled, and `localStorage.setItem( 'debug', 'wc-admin:tracks' );` is set to debug tracks.
3. Make sure browser console is opened.
2. Make sure you've set treatment for the experiment `woocommerce-tasklist-progression-headercard-10`.

**Testing the basic flow**
1. Go to WooCommerce > Home
2. Observe the task lists shown, it should have 6 tasks, it should all be in "incomplete" state and `products` task is selected by default
3. Click on either "Add products" button or the "Add my products" task
4. It should navigate to "Add products" task
5. View browser console, it should have `wcadmin_tasklist_click` and `wcadmin_task_view` tracks being recorded with parameter `task_name: products`
6. Add a product to complete the task
7. Go back to WooCommerce > Home
8. Observe the products task is completed and "Set up payments" task is selected
9. Make sure the header title is "Choose payment methods" since WCPay is not selected in setup
9. Complete the rest of the tasks
10. Choose "Keep list" when it prompts
10. Go to WooCommerce > Home, make sure "Personalize my store" is selected by default, with the CTA button in header becomes secondary (it's styled by white background with themed colored borders)

**Testing WCPay task**
1. If you're using a JN, you'd have to install [WCPay dev tools](https://github.com/Automattic/woocommerce-payments-dev-tools) and set WCPay in dev mode.
1. Go back to setup wizard and skip to Business Features step. Make sure to select WooCommerce Payments to be installed. Finish the setup
8. Go to WooCommerce > Home, observe "Set up payments" task is incomplete
9. Attempt to click on the task and make sure it doesn't navigate to anywhere unlike other tasks
9.  The header card should contain the title "It's time to get paid"
9. Click on "Verify details"
10. Observe it navigates to Jetpack or WCPay connect. Connect your site with WCPay
11. Navigate back to WooCommerce > Home, observe that the "Set up payments" task is completed
12. Attempt clicking the task, it should go to `payments` task

**Optional: Testing illustration theme styling**
1. Go to Users > All Users
2. Edit your user
3. Select a different admin color scheme and click on "Update Profile"
4. Go to WooCommerce > Home
5. Observe the illustration has its main colors changed to the selected theme
